app/controllers/donations_controller.rb

```diff
@@ -77,14 +77,27 @@ def show
 
   def update
     @donation = Donation.find(params[:id])
+    @original_source = @donation.source
     ItemizableUpdateService.call(itemizable: @donation,
       params: donation_params,
       type: :increase,
       event_class: DonationEvent)
+    flash.clear
+    flash[:notice] = "Donation updated!"
     redirect_to donations_path
   rescue => e
     flash[:alert] = "Error updating donation: #{e.message}"
-    render "edit"
+    load_form_collections
+    # calling new(donation_params) triggers a validation error if line_item quantity is invalid
+    @previous_input = Donation.new(donation_params.except(:line_items_attributes))
+    line_items = []
+    donation_params[:line_items_attributes].values.each { |attr|
+      attr.delete(:_destroy)
+      line_items.push(attr)
+    }
```

Comment:
I think this would put back any line items that the user removed?

Reply:
I think you would have been right, but I originally put this line to pass donation controller request rspec tests which sent the _destroy parameter. Those tests I found out are outdated, the way the donation form is currently implemented the _destroy parameter is never sent, the donation object is just overridden with the new line_items_attributes. updated those rspec tests and removed this line in commit [10fa617](https://github.com/rubyforgood/human-essentials/pull/4483/commits/10fa617f2fc03cb97caf8a1666a4911607ceb5d9)